src/ev3_bluetooth.c

```diff
@@ -0,0 +1,332 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <poll.h>
+#include <ctype.h>
+#include "../firmware_headers/bluetooth/bluetooth.h"
+#include "../firmware_headers/bluetooth/hci.h"
+#include "../firmware_headers/bluetooth/hci_lib.h"
+#include "../firmware_headers/bluetooth/rfcomm.h"
+
+#include "ev3_bluetooth.h"
+
+/**
+ * Channel 1 is already taken by the UI process of the robot, so we use 2.
+ */
+#define DEFAULT_C4EV3_RFCOMM_CHANNEL    2
+#define MAX_CONNECTIONS                 7
+#define SHORT_BLUETOOTH_ADDRESS_LENGTH  sizeof("000000000000")
+
+#define KNOWN_BLUETOOTH_NAMES_FILE_NAME_LENGTH 256
+#define KNOWN_BLUETOOTH_NAMES_FILE_ROW_LENGTH  256
+
+void BluetoothInit () { }
+
+BluetoothConnectionHandle connections[MAX_CONNECTIONS];
+int currentConnections = 0;
+
+void addConnectionToList (BluetoothConnectionHandle c) {
```

Comment:
Same. All internal functions should be static, so that:

* They don't pollute the global namespace
* You get warnings if they are unused
* They don't become API that client code relies on.

Reply:
I've fixed all the internal functions here 26c4aa8

I created different commits for the different fixes, hoping that it makes the review easier. Later I will squash the commits.